common/beerocks/bcl/source/network/get_iface_stats.cpp

```diff
@@ -0,0 +1,243 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include <bcl/network/network_utils.h>
+
+#include <easylogging++.h>
+
+#include <errno.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+#include <net/if.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#define IFLIST_REPLY_BUFFER 8192
+
+namespace beerocks {
+namespace net {
+
+/**
+ * @brief Netlink request type.
+ */
+struct nl_req_t {
+    /**
+     * Netlink message
+     */
+    struct nlmsghdr hdr;
+
+    /**
+     * "general form of address family dependent" message, i.e. how to tell which AF we are
+     * interested in. */
+    struct rtgenmsg gen;
+};
+
+/**
+ * @brief Gets interface statistics for the given network interface.
+ *
+ * Gets interface stats for given network interface from the specified Netlink message of type
+ * RTM_NEWLINK.
+ *
+ * @param[in] h Pointer to the Netlink message containing the data.
+ * @param[in] iface_name Name of the network interface.
+ * @param[in, out] iface_stats Interface statistics structure with values read.
+ *
+ * @return True on success and false otherwise.
+ */
+static bool get_iface_stats(const struct nlmsghdr *h, const std::string &iface_name,
+                            sInterfaceStats &iface_stats)
+{
+    bool result = false;
+
+    struct ifinfomsg *iface = static_cast<ifinfomsg *>(NLMSG_DATA(h));
+
+    size_t length = 0;
+    if (h->nlmsg_len > NLMSG_LENGTH(sizeof(*iface))) {
+        length = h->nlmsg_len - NLMSG_LENGTH(sizeof(*iface));
+    }
```

Comment:
`RTA_OK` will always fail in case `length=0` so there is no reason to get past here
```suggestion
    if (h->nlmsg_len > NLMSG_LENGTH(sizeof(*iface))) {
        return false;
    }
    size_t length = h->nlmsg_len - NLMSG_LENGTH(sizeof(*iface));
```

Reply:
Done